modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaBaseModelListenerCheck.java

```diff
@@ -55,18 +59,55 @@ protected String format(
 		return content;
 	}
 
-	private String _formatMethodDefinition(JavaMethod javaMethod) {
-		String javaMethodContent = javaMethod.getContent();
-
+	private String _formatMethod(JavaMethod javaMethod) {
 		String javaMethodName = javaMethod.getName();
 
-		if (!javaMethodName.equals("onAfterUpdate") &&
-			!javaMethodName.equals("onBeforeUpdate")) {
+		if (javaMethodName.equals("onAfterUpdate") ||
+			javaMethodName.equals("onBeforeUpdate")) {
+
+			return _formatMethodDefinitionWithParameterUpgrade(javaMethod);
+		}
+
+		return _formatMethodDefinitionWithoutParameterUpgrade(javaMethod);
+	}
 
+	private String _formatMethodDefinitionWithoutParameterUpgrade(
+		JavaMethod javaMethod) {
+
+		String javaMethodContent = javaMethod.getContent();
+
+		Matcher matcher = _superMethodPattern.matcher(javaMethodContent);
+
+		if (!matcher.find()) {
 			return javaMethodContent;
 		}
 
-		javaMethodContent = _formatSuper(javaMethodContent);
+		String methodCall = JavaSourceUtil.getMethodCall(
+			javaMethodContent, matcher.start());
+
+		methodCall = methodCall.trim();
+
+		String parameterName = JavaSourceUtil.getParameters(methodCall);
+
+		String cloneMethod = StringBundler.concat(
+			_extendedClassType, " original",
+			StringUtils.capitalize(parameterName), " = (", _extendedClassType,
+			")model.clone();");
+
+		String spacing = matcher.group(1);
+
+		String newJavaMethodContent = StringUtil.replace(
+			javaMethodContent, methodCall,
+			joinLines(cloneMethod, spacing + methodCall));
```

Comment:
Possible change: consider inlining the cloned model instead of making it a separate statement/variable.

```
@Override
public void onAfterRemove(t model) {
	super.onBeforeUpdate(model);
}

@Override
public void onAfterRemove(t model) {
	super.onBeforeUpdate((t)model.clone(), model);
}

@Override
public void onAfterRemove(t model) {
	t originalModel = (t)model.clone();

	super.onBeforeUpdate(originalModel, model);
}
```